src/s3.js

```diff
@@ -78,9 +79,46 @@ const moveObjects = async (objects, oldPrefix, newPrefix) => {
   console.log(`Reset completed. ${newPrefix} folder is now repopulated`);
 };
 
+const uploadObject = async (fileName, file) => {
+  const s3 = createS3Client();
+  // Read content from the file
+  const fileContent = fs.readFileSync(file);
+
+  // Setting up S3 upload parameters
+  const params = {
+    Bucket: `${S3_BUCKET_NAME}/test-staging`,
+    Key: fileName,
+    Body: fileContent,
+  };
```

Comment:
It's more common to keep the `Bucket` value only for the bucket name. Any folders we want to add should be in the `Key`. Additionally, instead of hardcoding `test-staging`, let's pass in a new variable into this function called `targetFolder`. Our new params should looks like this:

```javascript
  const params = {
    Bucket: S3_BUCKET_NAME,
    Key: `${targetFolder}${fileName}`,
    Body: fileContent,
  };
```

Reply:
With the new `targetFolder` parameter, it will cause upstream changes to `uploadFolderContents` and `send-it.js`. Then in your `send-it.js`, we can set `targetFolder` to either `staging/` or `test-staging/` and pass it into your function

Reply:
```mattq@ZOHAR90:~/dev/cat-bot$ npm run full-send

> cat-bot@1.0.0 full-send /home/mattq/dev/cat-bot
> node src/utils/send-it.js

Found file: tasty_chicken.jpg...
Found file: tommy_fighter.png...
File uploaded successfully: https://cat-bot-bucket.s3.amazonaws.com/undefinedtommy_fighter.png
File uploaded successfully: https://cat-bot-bucket.s3.amazonaws.com/undefinedtasty_chicken.jpg
```

The file paths got screwed up after I made those changes! I'll try to fix it at work tonight.

Reply:
I see the mistake already lol. I'm rushing because of work

Reply:
One last piece you're missing but let's see if you figure it out 😉

Reply:
You can optionally make this function more robust by adding this at the beginning:
```javascript
if (!targetFolder) {
  throw new Error(`Invalid targetFolder value: ${targetFolder}`);
}
```

It's also known as defensive programming. That will throw an error if `targetFolder` is any falsy value (e.g. `null`, `undefined` as some examples). That will prevent the code from continuing and hitting the S3 API

Reply:
Nice. Ok I added it!